packages/integrations/mdx/README.md

```diff
@@ -78,116 +78,98 @@ Visit the [MDX docs](https://mdxjs.com/docs/what-is-mdx/) to learn about using s
-
-Any additional plugins you apply in your MDX config will be applied *after* your configured Markdown plugins.
-
-#### `astroDefaults`
-
-Astro's MDX files will apply only [Astro's default plugins](/en/reference/configuration-reference/#markdownextenddefaultplugins), without inheriting the rest of your Markdown config. 
-
-This example will apply the default [GitHub-Flavored Markdown](https://github.com/remarkjs/remark-gfm) and [Smartypants](https://github.com/silvenon/remark-smartypants) plugins alongside [`remark-toc`](https://github.com/remarkjs/remark-toc) to your MDX files, while ignoring any `markdown.remarkPlugins` configuration:
-
-```js "extendPlugins: 'astroDefaults'"
+```ts
 // astro.config.mjs
+import { defineConfig } from 'astro/config';
+import mdx from '@astrojs/mdx';
 import remarkToc from 'remark-toc';
+import rehypeMinifyHtml from 'rehype-minify-html';
 
-export default {
-  markdown: {
-    remarkPlugins: [/** ignored */]
-  },
-  integrations: [mdx({
-    remarkPlugins: [remarkToc],
-    // Astro defaults applied
-    extendPlugins: 'astroDefaults',
-  })],
-}
+export default defineConfig({
+  integrations: [
+    mdx({
+      syntaxHighlight: 'shiki',
+      shikiConfig: { theme: 'dracula' },
+      remarkPlugins: [remarkToc],
+      rehypePlugins: [rehypeMinifyHtml],
+      remarkRehype: { footnoteLabel: 'Footnotes' },
+      githubFlavoredMarkdown: false,
+    })
+  ]
+})
 ```
 
-#### `false`
+:::caution
+MDX does not support passing remark and rehype plugins as a "string." You should install, import, and apply the plugin function instead.
+:::
 
-Astro's MDX files will not inherit any [`markdown` options](https://docs.astro.build/en/reference/configuration-reference/#markdown-options), nor will any Astro Markdown defaults be applied:
+📚 See the [Markdown Options reference](https://docs.astro.build/en/reference/configuration-reference/#markdown-options) for a complete list of options.
 
-```js "extendPlugins: false"
-// astro.config.mjs
-import remarkToc from 'remark-toc';
-
-export default {
-  integrations: [mdx({
-    remarkPlugins: [remarkToc],
-    // Astro defaults not applied
-    extendPlugins: false,
-  })],
-}
-```
+### `extendMarkdownConfig`
 
-### `remarkRehype`
+- **Type:** `boolean`
+- **Default:** `true`
 
-Markdown content is transformed into HTML through remark-rehype which has [a number of options](https://github.com/remarkjs/remark-rehype#options).
+MDX will extend [your project's existing Markdown configuration](https://docs.astro.build/en/reference/configuration-reference/#markdown-options) by default. All options will be **deeply merged,** including your remark / rehype plugins and your Shiki configuration.
 
-You can set remark-rehype options in your config file:
+For example, say you need to disable GitHub-Flavored Markdown and apply an extra remark plugin for MDX files. You can apply these options like so, with `extendMarkdownConfig` enabled by default:
```

Comment:
I'm wondering whether these examples might be better suited to the docs page itself? I don't think these tell me anything new. They visually show what it looks like when both Markdown and MDX have config options set, but I'm not sure that's needed here?

The docs page here will need updating anyway, to correspond to these changes:  [/#configuring-markdown-and-mdx](https://docs.astro.build/en/guides/markdown-content/#configuring-markdown-and-mdx)

Maybe what you have here as examples actually replaces what's there? Since *that's* where we show what it looks like when both have config settings are applied.

Reply:
Agreed! I'll put a docs PR on-hold until we get a dev approval here. May be some final API reshuffling.

Reply:
Alright, I think we have dev consensus. Moving to a docs PR!